examples/hardhat/test/counter.ts

```diff
@@ -20,12 +20,12 @@ describe('Counter', () => {
     const counterFactory = await ethers.getContractFactory('Counter', signers[0])
 
     counter = await counterFactory.deploy()
-    await counter.deployed()
+    await counter.waitForDeployment()
```

Comment:
@fvictorio  have a look at changes in this file

Reply:
LGTM, although I'd rather do just this:

```ts
    counter = await ethers.deployContract('Counter')
```

I'd also remove the `signers[0]` part because the first signer is used by default (unless you are being explicit on purpose).

Reply:
Good call. It's much simpler now. I want to keep these examples as a sort of "preferred usage" guide.